src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs

```diff
@@ -38,6 +53,31 @@ public ModalImporterPresenter(IModalImporterView view, IDataSetToDataRepositoryM
                {
                   var dataRepo = _dataRepositoryMapper.ConvertImportDataSet(d.DataSource, i++, pair.Key);
                   dataRepo.ConfigurationId = id;
+                  var molecule = dataRepo.ExtendedPropertyValueFor(moleculeName);
+                  var moleculeDescription = (metaDataCategories?.FirstOrDefault(md => md.Name == moleculeName)?.ListOfValues.FirstOrDefault(v => v.Key == dataRepo.ExtendedPropertyValueFor(moleculeName)))?.Value;
+                  var molecularWeightDescription = dataRepo.ExtendedPropertyValueFor(molWeightName);
+
+                  if (moleculeDescription != null)
+                  {
+                     if (string.IsNullOrEmpty(molecularWeightDescription))
+                     {
+                        molecularWeightDescription = moleculeDescription;
+                        dataRepo.ExtendedProperties.Add(new ExtendedProperty<string>() { Name = molWeightName, Value = moleculeDescription });
+                     }
+                     else
+                     {
+                        if (moleculeDescription != molecularWeightDescription)
+                           throw new InconsistenMoleculeAndMoleWeightException();
```

Comment:
Lines 69-70 are **string** comparison, right?
I wonder if it will not make some trouble (e.g. same value but different notations (e.g. `420 g/mol <=> 4.2E+2 g/mol`) or minimal numercal differences (e.g. `420 g/mol <=> 419,9999999` g/mol).
I think **numerical** comparison with some predefined **comparison tolerance** would be better

Reply:
@Yuri05 You are definitely right. I changed it already so now it used the parsed double values.